.github/workflows/release-python-package.yaml

```diff
@@ -0,0 +1,129 @@
+name: Release Python Package
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_name:
+        description: "One of major, minor, or patch"
+        required: true
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+        default: patch
+  pull_request:
+    branches:
+      - '**'
+
+env:
+  UV_SYSTEM_PYTHON: 1
+  DEFAULT_VERSION_NAME: patch
+
+jobs:
+  deploy-package:
+    runs-on: ubuntu-latest
+    name: ${{ github.event_name == 'pull_request' && '(Dry Run) ' || '' }}Publish Python Package to PyPI
+
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
+      - name: Setup Pixi Environment
+        uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          pixi-version: latest
+          cache: true
+          cache-write: ${{ github.event_name == 'push' }}
+
+      - name: Set up Python
+        run: |
+          uv tool install bump2version
```

Comment:
bump2version can be defined within the `pyproject.toml` under `[project.optional-dependencies]`?

Reply:
same comment for the other steps that uses `uv`?

Reply:
Ah, I see where you're going with the commentary -- that it may be better to just consolidate the deps (including stuff used for CI/CD publishing). Strictly speaking, I think it's better to have only runtime dependencies present in `pyproject.toml`, and the use of `optional-dependencies` is best for the situation where we break out the functionality + dependencies of our project at runtime. Historically it's been used for tests, docs, etc. but now that I've had many reps building packages, that way of configuration confuses runtime deps from development environment deps.